comp/otelcol/collector/def/component.go

```diff
@@ -0,0 +1,22 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package collector defines the OpenTelemetry Collector component.
+package collector
+
+import (
+	"context"
+
+	"github.com/DataDog/datadog-agent/comp/otelcol/otlp"
+)
+
+// team: opentelemetry
+
+// Component specifies the interface implemented by the collector module.
+type Component interface {
+	Start(ctx context.Context) error
```

Comment:
I don't think these need to be public on the interface. Instead the Lifecycle hooks are calling them.